### DISCUSSION TITLE
Accessing variables within conditional function

### DISCUSSION THREAD

**Discussion by wsr017** [jspsych/jsPsych#859] [Q&A] [answered]
Hello!

I am trying to make a simple break in my experiment after the participants complete a certain number of trials. There should be a break whenever `completed_trials` is a value contained within the `break_intervals` array. `completed_trials` is incremented on the finish of the previous trial. Here is what I have:

```
    var break_period = {
      type: 'html-keyboard-response',
      stimulus: "<p><b>BREAK</b></p><p>Good job! You have completed <b>" + completed_trials + "</b> out of <b>" + (nTrials) + " trials.</b></p><p>Take a break and relax your eyes. Press C when you want to continue the experiment. </p>",
      choices: [67],
    }

    var break_if_node = {
      timeline: [break_period],
      conditional_function: function(){
        if (break_intervals.includes(completed_trials)) {
          return true;
          } else {
          return false; 
        }
      }
    }
```

The problem that I'm having is that `completed_trials` always prints as 0 in the `break_period` stimulus. I know that `completed_trials` is incrementing correctly by checking in console and because the `break_if_node` evaluates to true at the desired intervals, but `completed_trials` is then printed as 0 in the stimulus. Is there something that I'm missing to pass the value of `completed_trials` along to `break_period`, so that people know how many trials they have done? 

This might not be a jsPsych specific question, but I'm pretty new to all of this.

Thanks so much for the help!

**Comment by vijaymarupudi** [comment, marked as answer]
When you're constructing the timeline, all the code is run when you are defining the timeline, not when the participant is running the experiment. If you'd like to display the 'current' information will the participant is running the experiment, you'll want to make the parameter a function <https://www.jspsych.org/overview/trial/#dynamic-parameters>

If you'd like to learn more, these are called closures in Javascript!

**Reply by wsr017** [reply]
Aha! Thank you so much, I must have missed that in the tutorial. This forum is awesome.
